bindings/python/src/OpenSpaceToolkitAstrodynamicsPy/GuidanceLaw/SequentialGuidanceLaw.cpp

```diff
@@ -0,0 +1,119 @@
+/// Apache License 2.0
+
+#include <OpenSpaceToolkit/Astrodynamics/GuidanceLaw/SequentialGuidanceLaw.hpp>
+
+using namespace pybind11;
+
+using ostk::core::container::Array;
+using ostk::core::type::Real;
+using ostk::core::type::Shared;
+using ostk::core::type::String;
+
+using ostk::mathematics::object::Vector3d;
+
+using ostk::physics::coordinate::Frame;
+using ostk::physics::time::Instant;
+using ostk::physics::time::Interval;
+
+using ostk::astrodynamics::GuidanceLaw;
+using ostk::astrodynamics::guidancelaw::SequentialGuidanceLaw;
+
+inline void OpenSpaceToolkitAstrodynamicsPy_GuidanceLaw_SequentialGuidanceLaw(pybind11::module& aModule)
+{
+    class_<SequentialGuidanceLaw, GuidanceLaw, Shared<SequentialGuidanceLaw>>(
+        aModule,
+        "SequentialGuidanceLaw",
+        R"doc(
+            A guidance law that sequences multiple guidance laws over specific time intervals.
+            
+            At each point in time, the applicable guidance law is selected and used to calculate
+            the thrust acceleration. Guidance laws don't need to be continuous, and can be added
+            in any order. If the instant does not fall within any of the intervals, the thrust
+            acceleration is zero. The guidance law intervals must not intersect each other.
+        )doc"
+    )
+
+        .def(
+            init<const Array<Interval>&, const Array<Shared<GuidanceLaw>>&>(),
+            arg("intervals") = Array<Interval>::Empty(),
+            arg("guidance_laws") = Array<Shared<GuidanceLaw>>::Empty(),
+            R"doc(
+                Constructor.
+
+                Args:
+                    intervals (List[Interval], optional): Array of intervals. Defaults to empty array.
+                    guidance_laws (List[GuidanceLaw], optional): Array of guidance laws. Defaults to empty array.
+
+                Raises:
+                    RuntimeError: If intervals and guidance laws arrays have different sizes, if any interval is undefined,
+                    if the guidance law is null or if the interval intersects with existing intervals.
+            )doc"
+        )
```

Comment:
```suggestion
            R"doc(
                Constructor.

                Args:
                    guidance_law_with_intervals: (list[tuple[GuidanceLaw, Interval]]): List of Guidance laws with their applicable intervals.

                Raises:
                    RuntimeError: If any interval is undefined, or if the guidance law is undefined. Also, if the interval intersects with other intervals.
            )doc"
        )
```
Sorry I know this might be a bit of a refactor but maybe this is a cleaner interface?
Alternatively, you can create a struct called GuidanceLawSequence or something which has a GuidanceLaw and an Interval.